ibis/expr/datatypes.py

```diff
@@ -248,8 +94,14 @@ def equals(self, other, cache=None):
     def _equal_part(self, other, cache=None):
         return True
 
-    def can_implicit_cast(self, other):
-        return self.issubtype(other)
+    def issubtype(self, parent):
+        return issubtype(self, parent)
+
+    def castable(self, target):
```

Comment:
Can you add a docstring here indicating that `castable` here means *implicitly* castable?

Reply:
Sure, I might be more verbose.